diff -up lua-5.3.3/src/lcode.c.manyconsts lua-5.3.3/src/lcode.c
--- lua-5.3.3/src/lcode.c.manyconsts    2016-05-13 19:09:21.000000000 +0000
+++ lua-5.3.3/src/lcode.c       2016-11-02 19:11:51.530772426 +0000
@@ -1017,8 +1017,8 @@ static void codeunexpval (FuncState *fs,
 */
 static void codebinexpval (FuncState *fs, OpCode op,
                            expdesc *e1, expdesc *e2, int line) {
-  int rk1 = luaK_exp2RK(fs, e1);  /* both operands are "RK" */
-  int rk2 = luaK_exp2RK(fs, e2);
+  int rk2 = luaK_exp2RK(fs, e2);  /* both operands are "RK" */
+  int rk1 = luaK_exp2RK(fs, e1);
   freeexps(fs, e1, e2);
   e1->u.info = luaK_codeABC(fs, op, 0, rk1, rk2);  /* generate opcode */
   e1->k = VRELOCABLE;  /* all those operations are relocatable */
